aptos-move/block-executor/src/proptest_types/types.rs

```diff
@@ -1063,224 +926,96 @@ pub(crate) enum GroupSizeOrMetadata {
-        // towards the block limit. We split the total into two pieces for these arguments.
-        // TODO: add variety to generating fee statement based on total gas.
-        FeeStatement::new(
-            self.total_gas,
-            self.total_gas / 2,
-            (self.total_gas + 1) / 2,
-            0,
-            0,
-        )
-    }
+// ID is just the unique index as u128.
+pub(crate) fn serialize_from_delayed_field_u128(value_or_id: u128, version: u32) -> Bytes {
+    let tuple = (value_or_id, version);
+    serialize_delayed_field_tuple(&tuple)
+}
 
-    fn output_approx_size(&self) -> u64 {
-        // TODO add block output limit testing
-        0
-    }
+pub(crate) fn serialize_from_delayed_field_id(
+    delayed_field_id: DelayedFieldID,
+    version: u32,
+) -> Bytes {
+    let tuple = (delayed_field_id.extract_unique_index() as u128, version);
+    serialize_delayed_field_tuple(&tuple)
+}
 
-    fn get_write_summary(
-        &self,
-    ) -> HashSet<
-        crate::types::InputOutputKey<
-            <Self::Txn as Transaction>::Key,
-            <Self::Txn as Transaction>::Tag,
-        >,
-    > {
-        HashSet::new()
-    }
+/// Serializes a (u128, u32) tuple into Bytes using BCS serialization
+fn serialize_delayed_field_tuple(value: &(u128, u32)) -> Bytes {
+    bcs::to_bytes(value)
+        .expect("Failed to serialize (u128, u32) tuple")
+        .into()
 }
 
-#[derive(Clone, Debug)]
-pub(crate) struct MockEvent {
-    event_data: Vec<u8>,
+/// The width of the delayed field is not used in the tests, and fixed as 8 for
+/// all delayed field constructions. However, only the real ID is actually
+/// serialized and deserialized (together with the version).
+pub(crate) fn deserialize_to_delayed_field_u128(bytes: &[u8]) -> Result<(u128, u32), bcs::Error> {
+    bcs::from_bytes::<(u128, u32)>(bytes)
 }
 
-impl TransactionEvent for MockEvent {
-    fn get_event_data(&self) -> &[u8] {
-        &self.event_data
-    }
+pub(crate) fn deserialize_to_delayed_field_id(
+    bytes: &[u8],
+) -> Result<(DelayedFieldID, u32), bcs::Error> {
+    let (id, version) = bcs::from_bytes::<(u128, u32)>(bytes)?;
+    Ok((DelayedFieldID::from((id as u32, 8)), version))
+}
 
-    fn set_event_data(&mut self, event_data: Vec<u8>) {
-        self.event_data = event_data;
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use test_case::test_case;
+
+    #[test_case((0u128, 0u32) ; "zero values")]
+    #[test_case((1u128, 42u32) ; "small values")]
+    #[test_case((u128::MAX, u32::MAX) ; "maximum values")]
+    #[test_case((12345678u128, 87654321u32) ; "large values")]
+    fn test_serialize_deserialize_delayed_field_tuple(tuple: (u128, u32)) {
+        // Serialize and then deserialize
+        let serialized = serialize_delayed_field_tuple(&tuple);
+        let deserialized =
+            deserialize_to_delayed_field_u128(&serialized).expect("Deserialization failed");
```

Comment:
I guess `unwrap()` is ok otherwise "Deserialization failed" does not give much info on why it is safe to unwrap 😅 or `claims::assert_ok!`?